src/test/ui/suggestions/dont-suggest-doc-hidden-variant-for-enum/dont-suggest-doc-hidden-variant-for-enum.rs

```diff
@@ -0,0 +1,7 @@
+// aux-build:serde.rs
+
+extern crate serde;
```

Comment:
I don't think that serde is a good name for the external crate here. It would lead to confusion with the actual crate serde. 

Also I'm not entirely sure what the point of this test is. There is no difference in your output than to the current output.

Something like this is more what I would expect of your code:

hidden_struct.rs
```rust
#[doc(hidden)]
pub struct Foo;
```

main.rs
```rust
fn main() {
    let _ = FOo;
}
```
output is:

```
error[E0425]: cannot find value `FOo` in this scope
 --> src/main.rs:5:13
  |
2 | pub struct Foo;
  | --------------- similarly named unit struct `Foo` defined here
...
5 |     let _ = FOo;
  |             ^^^ help: a unit struct with a similar name exists (notice the capitalization): `Foo`
```

Reply:
Try run test locally, after  #87158 it suggests `serde::__private::Some`

Reply:
> Try to run test locally, after #87158 it suggests serde::__private::Some.

My bad. I was testing stable.

> I don't see how your test case is related to issue.

There is a suggestion for a hidden struct/item. Isn't that what you are trying to suppress?

Reply:
@fellow554 if there is doc-hidden reexport of struct enum variant, I want suggestion to prefer path which doesn't lead to doc-hidden module. Like in test case.

I want the test to pass. I will change the name of `serde` to something else, but originally this is just minimal reproduction of issue I got with serde. They have "private" reexports.

I'm not sure where is `expected` expr is selected, to make compiler prefer paths which are not doc-hidden, I need some guidance.

Thanks!

Reply:
@hellow554 P.S. using local module doesn't work it should be external crate.

Reply:
I already found that root cause is actually here
https://github.com/rust-lang/rust/blob/8024983ea73e1d1c16720a424d39fd1c239eee68/compiler/rustc_middle/src/ty/print/pretty.rs#L325-L358